vllm/inputs/preprocess.py

```diff
@@ -241,14 +230,24 @@ def _extract_prompt_components(
                 request_id=request_id,
                 lora_request=lora_request,
             )
-            multi_modal_data = None
-            mm_processor_kwargs = None
-        elif parsed["type"] == "tokens":
-            prompt_text = None
+
+            return token_inputs(
+                prompt=prompt_text,
+                prompt_token_ids=prompt_token_ids,
+            )
+
+        if parsed["type"] == "tokens":
             prompt_token_ids = parsed["content"]["prompt_token_ids"]
             multi_modal_data = parsed["content"].get("multi_modal_data")
             mm_processor_kwargs = parsed["content"].get("mm_processor_kwargs")
```

Comment:
can we avoid repeat lookups for these? or does that pose a type checking issue
```suggestion
        parsed_type = parsed["type"]
        if parsed_type:
            content = parsed["content"]
            prompt_token_ids = content["prompt_token_ids"]
            multi_modal_data = content.get("multi_modal_data")
            mm_processor_kwargs = content.get("mm_processor_kwargs")
```

Reply:
Yeah the type checker doesn't like that.